http/src/controllers/users.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { DataRepository } from './../database/dataRepository';
+import { IncomingMessage, ServerResponse } from "http";
+import { User } from "../interfaces";
+import { parseRequestData, sendJsonResponse } from "../utils";
+
+export class UserController {
+  static userCache:Map<string, User> = new Map();
+
+  static createUser(req: IncomingMessage, res: ServerResponse) {
+    parseRequestData(req, (data) => {
+      const user = JSON.parse(data);
+      user.id = DataRepository.usersLength() + 1;
+      user.links.user = req.url;
+      DataRepository.addUser(user);
+      sendJsonResponse(res, 201, user);
+    });
+  }
+
+  static getAllUsers(req: IncomingMessage, res: ServerResponse) {
+    const responseUsers = DataRepository.getUsers().map(({ name, id, email }: User) => {
+      return { name, id, email }
+    })
```

Comment:
nit
this type of logic is a service responsibility.

Reply:
fixed it